mkosi/kmod.py

```diff
@@ -79,19 +148,38 @@ def filter_firmware(
         firmwared = Path("usr/lib/firmware")
         with chdir(root):
             all_firmware = set(firmwared.rglob("*"))
+
+        patterns = [p[3:] for p in include if p.startswith("re:")]
         regex = re.compile("|".join(include))
+
         for f in all_firmware:
             rel = os.fspath(Path(*f.parts[3:]))
-            if regex.search(rel):
+            if (patterns and regex.search(rel)) or globs_match_firmware(rel, globs):
                 firmware.add(f)
 
+    logging.debug(f"Including {len(firmware)} firmare files")
+
     return firmware
 
 
 def normalize_module_name(name: str) -> str:
+    # Replace '_' by '-'
```

Comment:
We generally should have more comments, but this one might be a tad too much. Maybe a few more comments could be added to the slightly less obvious function below instead? :)